client/state/data-layer/wpcom-http/freshness-verifier.js

```diff
@@ -0,0 +1,69 @@
+export const CLEAN_OLDER_THAN = 3 * 60 * 1000;
+
+/** @type {Date} holds timestamp of the last time a cleaning operation was performed */
+let lastCleaningTime;
+
+/** @type {Map} holds request action keys, and last execution times */
+const history = new Map();
+
+/**
+ * Generate a deterministic key for comparing request actions
+ *
+ * @param   {Object} action redux action
+ * @returns {String}        action key
+ */
+const buildActionKey = ( action ) => JSON.stringify(
+	action.meta ? { ...action, meta: undefined } : action
+);
```

Comment:
or we send in a fake dispatcher and trap those HTTP calls, since it should dispatch an `HTTP_REQUEST` action. of course, then we couldn't prevent other non-HTTP initiating actions from taking place.

good point you make. 

> force users to pass an efficient key builder using what they know about the action

would love to make it so we don't have to think about this part. it's not something I think people should need to know about. it should "just work"

how about for now we leave in your key-builder which serializes the whole action without `meta` and continue to think about this problem